src/protos/mod.rs

```diff
@@ -339,6 +341,42 @@ pub mod temporal {
                     }
                 }
 
+                impl From<workflow_commands::RequestCancelActivity> for command::Attributes {
+                    fn from(c: workflow_commands::RequestCancelActivity) -> Self {
+                        Self::RequestCancelActivityTaskCommandAttributes(
+                            RequestCancelActivityTaskCommandAttributes {
+                                scheduled_event_id: c.scheduled_event_id,
+                            },
+                        )
+                    }
+                }
+
+                #[derive(derive_more::Display, Debug)]
+                pub enum CancellationTypeConversionError {
+                    InvalidCancellationType,
+                }
+
+                impl TryFrom<i32> for ActivityCancellationType {
```

Comment:
You don't need this -- `from_i32` already exists on these enums

Reply:
great, happy to get rid of this ugly piece

Reply:
Yes, there are a handful of examples of us already doing this in a few different places. Just search for usages of the `from_i32` fn.